master.js

```diff
@@ -2673,22 +2673,32 @@ var polling = bot.adapter.in = {
 	},
 
 	openSocket : function ( url, discard ) {
-		//chat sends an l query string parameter. seems to be the same as the
-		// since xhr parameter, but I didn't know what that was either so...
-		//putting in 0 got the last shitload of messages, so what does a high
-		// number do? (spoiler: it "works")
-		var socket = new WebSocket( url + '?l=99999999999' );
-
-		if ( discard ) {
-			socket.onmessage = function () {
-				socket.close();
-			};
-		}
-		else {
-			this.socket = socket;
-			socket.onmessage = this.ondata.bind( this );
-			socket.onclose = this.socketFail.bind( this );
+		var socket;
+		// socket-saver extension specific code:
```

Comment:
`openSocket` is called (eventually) from `/join` to, well, join a room. That's why there's the `discard` option...

So we either fix up `/join` to do something else or change `openSocket` or its caller to match the new behaviour.